### ISSUE TITLE
vnl_matrix_fixed operator() implementation: .hxx include necessary

### ISSUE THREAD

**Issue by jmon12** [vxl/vxl#775]
#### TLDR

To use `vnl_matrix_fixed<double, u1, u2>::operator()`, the user has to include the `vnl_matrix_fixed.hxx` implementation file, not only the `vnl_matrix_fixed.h` header.

#### Context

I'm currently migrating from vxl-1.18.0 to vxl-2.0.2. After some very minor tweaks of my code (great job by the way, very smooth migration), the linking fails.

After some investigation, one of the only notable differences for `vnl_matrix_fixed` from 1.18 to 2.0.2 is that the implementation of the operator() (for my specific case) has been moved to the corresponding `vnl_matrix_fixed.hxx` implementation file.

### Steps to Reproduce

The following code will link with vxl-1.18, but not with vxl-2.0.2:
```
#include <vnl/vnl_matrix_fixed.h>

int main()
{
	vnl_matrix_fixed<double,8,3> coords;
	coords(1,2) = 1;

	return 0;
}
```

Including the implementation header fixes the problem, i.e.
```
#include <vnl/vnl_matrix_fixed.h>
#include <vnl/vnl_matrix_fixed.hxx>

int main()
{
	vnl_matrix_fixed<double,8,3> coords;
	coords(1,2) = 1;

	return 0;
}
```

To compile:
```
$CXX -std=c++11 -Ipath/to/vxl/include/vxl/core -Ipath/to/vxl/include/vxl/vcl -Lpath/to/vxl/lib/ -lvnl main.cpp -o out
```

### Expected behavior

`#include <vnl/vnl_matrix_fixed.h>` should be enough for using the () operator.

### Actual behavior

Without `#include <vnl/vnl_matrix_fixed.hxx>`, the linking fails:
```none
Undefined symbols for architecture x86_64:
  "vnl_matrix_fixed<double, 8u, 3u>::operator()(unsigned int, unsigned int)", referenced from:
      _main in main-8e69f9.o
ld: symbol(s) not found for architecture x86_64
```

### Reproducibility

Always.

### Versions

As described: 
 - VXL 1.18.0 is not problematic
 - VXL 2.0.2 leads to the problem.

VXL has been compiled with the same compiler on the machine compiling the aforementioned code.
Note that only CORE_NUMERICS is built.

### Environment

The described case has been tested and reproduced on two OS:
 - MacOS with *cmake 3.17.1* and *clang++ Apple LLVM version 10.0.1 (clang-1001.0.46.4)*
 - CentOS7 with *cmake 3.14.6* and *gcc 5.3.1*

**Comment by hjmjohnson** [comment]
@jmon12 see #776 for fix.

**Comment by jmon12** [comment]
Thank you!
